privacy_consent/models/privacy_consent.py

```diff
@@ -0,0 +1,207 @@
+# -*- coding: utf-8 -*-
+# Copyright 2018 Tecnativa - Jairo Llopis
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+import hashlib
+import hmac
+
+from odoo import api, fields, models
+
+
+class PrivacyConsent(models.Model):
+    _name = 'privacy.consent'
+    _description = "Consent of data processing"
+    _inherit = "mail.thread"
+    _rec_name = "partner_id"
+    _sql_constraints = [
+        ("unique_partner_activity", "UNIQUE(partner_id, activity_id)",
+         "Duplicated partner in this data processing activity"),
+    ]
+
+    active = fields.Boolean(
+        default=True,
+        index=True,
+    )
+    accepted = fields.Boolean(
+        track_visibility="onchange",
+        help="Indicates current acceptance status, which can come from "
+             "subject's last answer, or from the default specified in the "
+             "related data processing activity.",
+    )
+    last_metadata = fields.Text(
+        readonly=True,
+        track_visibility="onchange",
+        help="Metadata from the last acceptance or rejection by the subject",
+    )
+    partner_id = fields.Many2one(
+        "res.partner",
```

Comment:
As you put one argument per line, it's better to explicitly name its argument name (`comodel_name`, `string`...)